src/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -164,78 +162,89 @@ [FromQuery] string dataType
                 );
             }
 
-            if (appLogic)
+            int existingElements = instance.Data.Count(d => d.DataType == dataTypeFromMetadata.Id);
+            if (dataTypeFromMetadata.MaxCount > 0 && existingElements >= dataTypeFromMetadata.MaxCount)
             {
-                return await CreateAppModelData(org, app, instance, dataType);
+                return Conflict(
+                    $"Element type `{dataType}` has reached its maximum allowed count ({dataTypeFromMetadata.MaxCount})"
+                );
```

Comment:
@danielskovli was this the error condition that was missing before? Can we make the response type `ProblemDetails` in that case?

Reply:
Yeah this is the check we discussed a while back 👍 

I think we can make the response whatever you/we want. I only did a 409 because it sort-of-almost followed the established pattern from this controller.

The frontend is specifically catching this response to display something other than "Oooops", but that check can easily be rewritten. It's not perfect as-is anyway, because it it assumes all 409 responses on `Create` is because of maxcount: https://github.com/Altinn/app-frontend-react/blob/feature/subform/src/features/subformData/useSubformMutations.tsx#L33

Reply:
Yeah I think 409 is good, I was only thinking about the string value being converted to `ProblemDetails` as we have middleware that annotates the ASP.NET Core trace with additional information when the response type is `ProblemDetails` (which means easier debugging)

Reply:
@martinothamar Unsurprisingly I don't know exactly what this should look like. But I can confirm that the frontend does check the body payload at all, only the response status code.

So if you're feeling inspired, please make whatever changes you think are necessary :)

Reply:
Pushed a commit for this

Reply:
Thanks!